### ISSUE TITLE
allow comma dangle to ignore single elements

### ISSUE THREAD

**Issue by rijnhard** [eslint/eslint#6400]
**What version of ESLint are you using?** 2.9

**What parser (default, Babel-ESLint, etc.) are you using?** default

**Please show your full configuration:** Not Required `'comma-dangle'  : [2, 'always-multiline'],`

**What did you expect to happen?**
if an array or an object is a single element, then do not require a dangling comma

```javascript
var obj = {
    keys: {
         t1: true, 
         t2: true,
         t3: true, //require this comma
    }, //**do not** require this comma
};
```

**What actually happened? Please include the actual, raw output from ESLint.**
`ESLint: error Missing trailing comma  (comma-dangle)`

**Comment by platinumazure** [comment]
Are you hoping to *not require* the comma for single-element cases, or *forbid* it? (Though honestly, I wonder if we should just allow both possibilities.)

**Comment by rijnhard** [comment]
@platinumazure wow that was fast. 

both probably for different use cases. by default it requires it. 
I would like to forbid it

**Comment by rijnhard** [comment]
Also just to clarify, this option should also be available for arrays, and not just objects to be comprehensive. 

As for individually controlling the settings for arrays or objects: I don't care for that, but you may have more insight on if it's overkill or should be allowed.

**Comment by platinumazure** [comment]
Any suggestions on how to configure it? (Could be kind of tricky since right now there's just a single string option, but do feel free to think outside the box!)

**Comment by rijnhard** [comment]
I'm going to sleep on it and will; get back to you tomorrow.

**Comment by rijnhard** [comment]
taking inspiration from key-spacing 

```javascript
var config = {
    'key-spacing': [2, {
        singleLine: {
            beforeColon: false,
            afterColon : true
        },
        multiLine: {
            beforeColon: false,
            afterColon : true,
            align      : 'colon'
        }
    }]
};
```

This is what pops to mind: 

```javascript
//sectioned config 
//no BC, most flexible, easiest to read

//multi is only with multiple elements, 
//as opposed to multiline which is regardless of number of elements

var config = {
    'comma-dangle': [2, {
        singleLine: never(default)|always|multi,
        multiLine:  never(default)|always|multi
    }]
};

var config = {
    'comma-dangle': [2, {
        singleLine: {
            array: never(default)|always|multi,
            object: never(default)|always|multi
        },
        multiLine: {
            array: never(default)|always|multi,
            object: never(default)|always|multi
        }
    }]
};
```

or 

```javascript
//override based config 
//most BC

var config = {
    'key-spacing': [2, "never|always|always-multiline|only-multiline", {
        //overrides
        //multi is only when multiple elements
        array: never|always|multi, 
        object: never|always|multi
    }]
};
```

**Comment by alberto** [comment]
Thanks for your interest in improving eslint. Unfortunately, it looks like consensus couldn't be reached on this issue and so I'm closing it. While we wish we'd be able to accommodate everyone's requests, we do need to prioritize. We've found that issues failing to reach consensus after 21 days tend never to reach consensus, and as such, we close those issues. This doesn't mean the idea isn't interesting, just that it's not something the team can commit to.
